### ISSUE TITLE
Conversation Item order is not correct

### ISSUE THREAD

**Issue by mrThirsty** [Microsoft/BotFramework-WebChat#1389]
The items in the webchat are not showing in the correct order.

The last user message will always be at the bottom of the chat, even after a reply has been sent by the bot.

 example conversation:

User: Hi
Bot: Hi how are you?
User: I am good thanks.
Bot: How can I help?

would display as such in the webchat window:

Bot:Hi how are you?
User: Hi
Bot: How Can I help?
User: I am good thanks.

You can actually see the users message in the above conversation being moved down below the bot reply after it has been displayed in the webchat window.

I can confirm that this is not an issue with the bot as the conversation order is not altered after a reply is received in the Emulator and in the "Test in Web" in the Azure Bot Channel details.

I have also put a very basic bot in that just echo's what the user said and the same issue happens.

Below is the configuration in JavaScript of the WebChat window:

`$(document).ready(function() {
        var userId = "d8e83df9-9eaa-e811-814c-5065f38b1651";
            
        var user = { id: userId, name: "Hindrajh Singh" };
        var bot = { id: "BOT" };

        (async function(){
            const res = await fetch("https://someendpoint.azurewebsites.net/api/token", { method: 'POST', 
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            },
            body: JSON.stringify({ UserID: userId }) });
            const { token } = await res.json();

            var botConnection = new window.WebChat.createDirectLine({
                token: token,
                user: user
            });

            window.WebChat.renderWebChat({
                directLine: botConnection,
                botAvatarInitials: 'BM',
                userAvatarInitials: 'HS',
                bot: bot,
                user: user,
                userID: userId,
            }, document.getElementById('BotChatGoesHere'));
        })().catch(err => console.error(err));
    });`

**Comment by corinagum** [comment]
The order of activities is determined by the timestamp on the DirectLine Service -- we don't use the conversationally natural order of activities.
Can you check your Network tab in developer tools for activity.timestamp, and verify that these are in time order?

**Comment by compulim** [comment]
This is the [portion of code](https://github.com/Microsoft/BotFramework-WebChat/blob/master/packages/core/src/reducers/activities.js#L25) that find the right position to insert the activity into. I put a lot of comments to make it more readable.

Do you think the logic here is correct or not?

**Comment by corinagum** [comment]
Closing due to inactivity

**Comment by compulim** [comment]
Hi @mrThirsty,

Could you review the code over here, https://github.com/Microsoft/BotFramework-WebChat/blob/master/packages/core/src/reducers/activities.js#L25?

This is the place where we upsert the activity into our UI. Do you think the logic is having issues over there?

**Comment by mrThirsty** [comment]
@compulim 

So I have done some more testing looking at the function you pointed me to. It is sorting the messages fine and inserting correctly so I have stepped back and looked at the page on the whole to see if I can spot something.

We are embedding this widget in a Microsoft Dynamics Portal, which auto generates a lot of stuff such as base page, navigation etc, and also has its own CSS and JS files. there is a default js file, in our case it is called default-1033.bundle.js, -1033 due to location and languages enabled on the Portal instance.

This JS file is getting loaded after the JS we have that configures the bot. I have managed to clone the page to my local machine and when I move this default.bundle.js to load before I load in the webchat.js and my JS that configs the bot runs then the issue doesn't happen.

So I can only surmise that there is an event in default-1033.bundle.js that is messing with your webchat.js.

The problem I have is that I can't see what it is, nor do we have control over where or when the default-1033.bundle.js is loaded. I have looked and there seems to be no way for me to get the webchat.js to load after this file has loaded, it will always load default-1033.bundle.js last.
